formspecs.lua

```diff
@@ -21,7 +21,9 @@ function travelnet.show_message(pos, player_name, title, message)
 			S("Exit"),
 			minetest.pos_to_string(pos)
 		)
-	minetest.show_formspec(player_name, travelnet_form_name, formspec)
+	minetest.after(0.1, function ()
```

Comment:
```suggestion
	minetest.after(0, function ()
```
To make it more clear that action is meant to be executed immediately on next globalstep and adding delay is not really intended.

Reply:
I tried this, it still fails sometimes if it's called this soon after the form closes. Will test it again to confirm, but the 0.1 value was chosen intentionally.

Reply:
> It works like 10% of the time with a delay of 0, it seems to work 100% of the time with a delay of 0.1

Sounds like delay is not really good fix and it seems to depend on some race condition, basically either client or server lag...
Could try if clearing formspec (sending empty formspec) first and after that sending actual intended formspec affects behavior in any way.

One way to actually fix issue (without fixing underlying engine issue) would be to not use exit buttons at all but instead replace previous formspec directly without asking client to close it first.

Reply:
Yeah I think removing exit buttons is the real fix, but using a delay causes minimal changes to the code and make sure it works in all circumstances, in case someone decides to use a exit button again.

I'm happy  to fix this properly, just letting you know it's going to be harder to maintain

Reply:
Is there a way to reject a formspec element using luacheck?

Reply:
> Is there a way to reject a formspec element using luacheck?

No, luacheck is not meant to do checks like that.
Mineunit can be used for automated tests to check actual behavior, I think no direct support for formspecs yet but can be done with simple override.

For maintenance, simply document it with simple comment. Then in case someone decides to use an exit button again someone for sure notices it.

Reply:
> For maintenance, simply document it with simple comment.

There would have to be quite a few, and that would not stop anyone from doing it in any meaningful way. We would have to catch this kind of thing by just reviewing PRs

Reply:
I've removed all the exit buttons, but I've had to add another delay on the edit_box call. There is also another delay already on the edit_formspec call, I think I will need to rewrite the way this works completely to get rid of these - it's possible and should be done, but it expands the scope of this issue significantly.

Reply:
@S-S-X I've reverted the exit button changes as I wasn't happy with them, better to go back to a working version ans leave it as an option.

I have reworked the way the formspec works in the PR to solve this issue with minimal changes, and without requiring timeouts https://github.com/mt-mods/travelnet/pull/35

Reply:
Resolving this as we should now focus on #35 , leaving this open however as it may be decided that #35 is too complex and this solution is prefered.